app/src/app/components/Map.tsx

```diff
@@ -36,6 +37,12 @@ export const MapComponent: React.FC = () => {
   };
   useEffect(fitMapToBounds, [mapOptions.bounds]);
 
+  useEffect(() => {
+    map.current?.on('mouseout', () => {
+      setTimeout(() => setHoverFeatures([]), 250);  
```

Comment:
H: Timeout approach seems to work here, but this should be in `mapEvents.ts` > `handleMapMouseOut. Unless this needs to rely on the component lifecycle?